frontend/app/components/language-switcher.tsx

```diff
@@ -14,11 +14,10 @@ export function LanguageSwitcher({ ...props }: LanguageSwitcherProps) {
   const { pathname } = useLocation();
   const { i18n, t } = useTranslation('common');
 
-  const altLang = i18n.language === 'fr' ? 'en' : 'fr';
-  const linkPath = pathname.replace(i18n.language, altLang);
+  const to = `${pathname}?lang=${i18n.language === 'fr' ? 'en' : 'fr'}`;
 
   return (
-    <Link {...props} to={linkPath} onClick={() => i18n.changeLanguage(altLang)}>
+    <Link to={to} reloadDocument {...props}>
```

Comment:
Spread first than override.

```suggestion
    <Link {...props} to={to} reloadDocument={true}>
```

Reply:
In this case it doesn't matter because of `type LanguageSwitcherProps = Omit<LinkProps, 'to' | 'reloadDocument'>`, but I will change it anyway for consistency with other code that might follow suit.

Reply:
[Fixed](https://github.com/DTS-STN/canadian-dental-care-plan/pull/17/files#diff-5a4ca5b528aaf8411c9cb8f6c6ba138f324ef5ae6d39537f8c6a6c2e204e78a9R20)